rules/Arguments/Rector/ClassMethod/ArgumentAdderRector.php

```diff
@@ -92,17 +93,11 @@ public function someMethod($value = true)
         );
     }
 
-    /**
-     * @return array<class-string<Node>>
-     */
     public function getNodeTypes(): array
     {
         return [MethodCall::class, StaticCall::class, ClassMethod::class];
     }
 
-    /**
-     * @param MethodCall|StaticCall|ClassMethod $node
-     */
     public function refactor(Node $node): MethodCall | StaticCall | ClassMethod | null
     {
         $this->haveArgumentsChanged = false;
```

Comment:
This actually disables PHPStorm autocomplete :/

Reply:
for which version of PHPStorm ?

Reply:
@TomasVotruba from what @ondrejmirtes said (in mentioned discussion) it works with newest PHPStorm.

Reply:
This was the issue last couple of years with similar refacorings. 

What version have you tested in successfuly? Please share a gif with working autocomplete without param annotations, I'm curious :)

Also, I'm not sure we can enforce new PHPStorm version requirement to the Rector codebase so early.

Reply:
Note that keeping the `@param` should not trigger errors in phpstan AFAICT (once you add the appropriate `@extends`). So you could keep the phpdoc for IDEs not having proper generic support yet.

Reply:
Currently I'm on 2022.3.2 Preview, but I believe it works also on earlier versions. The thing is PHPStorm can't resolve generic phpDoc if there's specific phpDoc in the implemented method (even if it's only `@return`):

![rector](https://user-images.githubusercontent.com/600668/213519439-d02e29eb-e4a2-4210-a70a-6cf904605994.gif)

I've tried `@inheritdoc` but it does not work either. Any ideas? 🙂

Reply:
@stof PHPStorm autocompletion only works with abstract `AbstractRector::refactor()` method... 😩

Reply:
This should be reported as a bug in PHPStorm.

Reply:
I would say both

Reply:
Created [WI-70906](https://youtrack.jetbrains.com/issue/WI-70906/PHP-Generics-info-is-missing-with-complex-inheritance).